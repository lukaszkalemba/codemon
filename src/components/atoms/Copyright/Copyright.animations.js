const getCopyright = animate => {
  return {
    variants: {
      animate: {
        opacity: 0.8,
        y: 0,
        transition: { duration: 0.8 },
      },
      initial: { opacity: 0, y: 20 },
    },
    animate,
    initial: 'initial',
  };
};

export default { getCopyright };
