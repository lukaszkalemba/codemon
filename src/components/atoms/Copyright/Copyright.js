import React, { useEffect } from 'react';
import { useInView } from 'react-intersection-observer';
import { useAnimation } from 'framer-motion';
import animations from './Copyright.animations';
import S from './Copyright.styles';

const Copyright = () => {
  const animation = useAnimation();

  const [copyrightRef, inView] = useInView({
    triggerOnce: true,
    threshold: 0.5,
  });

  useEffect(() => {
    if (inView) {
      animation.start('animate');
    }
  }, [animation, inView]);

  const copyrightAnimations = animations.getCopyright(animation);
  return (
    <S.Copyright ref={copyrightRef} {...copyrightAnimations}>
      <span>lukaszkalemba&copy; 2020.&nbsp;</span>
      <span>Wszelkie prawa zastrze≈ºone.</span>
    </S.Copyright>
  );
};

export default Copyright;
