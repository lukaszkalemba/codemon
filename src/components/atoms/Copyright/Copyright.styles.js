import styled from 'styled-components';
import { motion } from 'framer-motion';

const Copyright = styled(motion.p)`
  font-size: ${({ theme }) => theme.font.size.body.xs};
  opacity: 0.8;
  padding-top: 30px;
  display: flex;
  flex-direction: column;
  margin: 5px 0;
  span {
    margin: 2px 0;
  }

  ${({ theme }) => theme.media.phone`
    flex-direction: row;
  `}

  ${({ theme }) => theme.media.laptop`
    padding-top: 0;
  `}
`;

export default { Copyright };
