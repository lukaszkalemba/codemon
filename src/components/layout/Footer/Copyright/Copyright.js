import React from 'react';

const Copyright = () => <p>codemon&copy;2020. Wszelkie prawa zastrze≈ºone.</p>;

export default Copyright;
