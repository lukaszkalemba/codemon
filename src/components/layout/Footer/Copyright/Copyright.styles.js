import styled from 'styled-components';

const Paragraph = styled.div`
  font-size: ${({ theme }) => theme.font.size.xxs};
  opacity: 0.4;
`;

export default { Paragraph };
