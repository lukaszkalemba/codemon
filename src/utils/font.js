export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
