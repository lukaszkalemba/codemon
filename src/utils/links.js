export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
