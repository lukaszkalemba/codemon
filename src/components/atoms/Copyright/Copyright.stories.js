import React from 'react';
import Copyright from './Copyright';

export default {
  title: 'Atoms/Copyright',
  component: Copyright,
};

const Template = args => <Copyright {...args} />;

export const Default = Template.bind({});
Default.args = {};
