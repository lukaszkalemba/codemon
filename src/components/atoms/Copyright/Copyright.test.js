import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import Copyright from './Copyright';

describe('<Copyright />', () => {
  it('matches snapshot', () => {
    const { container } = render(<Copyright />);

    expect(container).toMatchSnapshot();
  });
});
