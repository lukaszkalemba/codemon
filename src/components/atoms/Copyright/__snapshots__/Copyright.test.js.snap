// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Copyright /> matches snapshot 1`] = `
<div>
  <p
    class="sc-bdnylx bRlJwW"
    style="opacity: 0; transform: translateY(20px) translateZ(0);"
  >
    <span>
      lukaszkalemba© 2020. 
    </span>
    <span>
      Wszelkie prawa zastrzeżone.
    </span>
  </p>
</div>
`;
